argilla/src/argilla/records/_dataset_records.py

```diff
@@ -494,3 +517,52 @@ def _infer_record_from_mapping(
             responses=responses,
             _dataset=self.__dataset,
         )
+
+    def _parse_suggestion_from_mapping(
+        self, data: Dict[str, Any], mapping: Dict[str, Tuple[Optional[str]]]
+    ) -> List[Suggestion]:
+        """Converts a mapped suggestion dictionary to a Suggestion object for use by the add or update methods.
+        Suggestions can be defined across multiple source values and mapped to single questions.
+        Args:
+            data: A dictionary representing the suggestion.
+            mapping: A dictionary mapping from source data keys/ columns to Argilla fields, questions, ids, etc.
+        Returns:
+            A Suggestion object.
+
+        """
+        suggestion_values = defaultdict(dict)
+
+        for (attribute_name, sub_attribute), source_key in mapping.items():
+            value = data.get(source_key)
+            schema_item = self.__dataset.schema.get(attribute_name)
+            suggestion_values[attribute_name].update(
+                {sub_attribute: value, "question_name": attribute_name, "question_id": schema_item.id}
```

Comment:
```suggestion
                {sub_attribute: value, "question_name": attribute_name, "question_id": schema_item.id}
```
maybe `sub_attribute` is not very specific and can be `attribute_value_key` or something?

Reply:
sub_attribute as yet only refers to the parameters of a suggestion. So score and agent. I don't think `_key` explains that, but I'm open to suggestions.

Reply:
@frascuchon I've reimplemented the PR using a class as suggested.